libmscore/segment.cpp

```diff
@@ -482,6 +482,9 @@ void Segment::add(Element* el)
       Q_ASSERT(track != -1);
       Q_ASSERT(el->score() == score());
       Q_ASSERT(score()->nstaves() * VOICES == int(_elist.size()));
+      // make sure offset is correct for staff
+      if (el->isStyled(Pid::OFFSET))
+            el->setOffset(el->propertyDefault(Pid::OFFSET).toPointF());
```

Comment:
If scaling is needed for offset, it should probably be applied not only in `Segment::add()`. For example, the following scenario still works incorrectly:
1) Make a small staff
2) Add there a staff text.
Its offset gets scaled (`y = -1.40sp` while the style default is `-2sp`)
3) Flip placement (<kbd>X</kbd>)
4) Flip placement again
Now offset equals to `-2sp` so it is not scaled again (actually scaling gets broken on step 3 already).

By the way, shifting offset a bit and resetting it to style default correctly assigns the scaled value, it looks like flipping placement direction avoids calling `styleValue()` but takes somehow the value directly from the style.

Reply:
I think I have it figured out, hopefully have a new commit soon...

Reply:
Done.  The original code should have worked for most non-text elements, because they generally explicitly set offset in this same basic way during layout.  But TextBase::layout() has this commented out (with a TODO that it should already be done).  That's something I've run into before and never quite understood.  But, no problem,  I just fixed the code in undoChangeProperty() where we change offset based on placement.